### PR TITLE
Only show a git revision in --version if it is an emscripten version

### LINKED ISSUE #10660: emcc --version reports git commit of emsdk checkout, not emscripten
Running `emcc --version` returns something like this:

```
emcc (Emscripten gcc/clang-like replacement) 1.39.8 (commit 86a4766a617150f44158ee63977d6c24b305a4ac)
Copyright (C) 2014 the Emscripten authors (see AUTHORS.txt)
This is free and open source software under the MIT license.
There is NO warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

However if emscripten is installed as a download through emsdk such as `emsdk install tot`, the "commit" bit will refer to the enclosing `emsdk` git checkout -- for instance `86a4766a617150f44158ee63977d6c24b305a4ac` is the currently latest update to emsdk, not the emscripten version that was installed.

I've found this very confusing a few times, thinking that I'm on the same build between multiple machines when in fact they're running different builds but have both run '`git update`' on emsdk.

Probably either the actual commit number of the emscripten checkout should be frozen into the build, or no commit should be displayed if the git repository is farther out in the filesystem than the enclosing directory of `emcc`.

### PR DESCRIPTION
**Pull request by kripken**
Noticed by @brion - this was indeed confusing! `git log` will look higher
up to a parent directory that is a git repo, so we would show that revision
which may be an emsdk revision or something else if the user has an
emscripten copy in their own repo.

Fixes #10660

### CODE REVIEW

File: emcc.py [hunk 1]
```diff
@@ -602,10 +602,15 @@ def run(args):
...
 
   if '--version' in args:
-    try:
-      revision = run_process(['git', 'show'], stdout=PIPE, stderr=PIPE, cwd=shared.path_from_root()).stdout.splitlines()[0]
-    except Exception:
-      revision = '(unknown revision)'
+    # if the emscripten folder is not a git repo, don't run git show - that can
+    # look up and find the revision in a parent directory that is a git repo
+    if not os.path.exists(shared.path_from_root('.git')):
+      revision = 'unknown revision'
```
Comment (sbc100) [inline emcc.py]: I'd rather invert the condition and branches.

Comment (sbc100) [reply emcc.py]: And invert the comment too `If the emscripten directory contains a .git, attempt to use a git show command to extract a version`.

**Comment by kripken** [comment]
Simplified as suggested.

I think @brion was suggesting something more - @brion correct me if I'm wrong - that if we are not a git checkout, then we are likely an emsdk checkout, and can figure out the emscripten revision from the version. That is, [look up the version in DEPS and find the hash for the emscripten repo.so forth](https://github.com/emscripten-core/emscripten/blob/master/docs/process.md#packaging-emscripten). That would be a bunch more work but maybe it would be worth it?

**Comment by brion** [comment]
Yes, I was thinking it'd be great to know the commit number when working with a tip of tree snapshot installed from emsdk. Would make it much easier to track down where in a git checkout I should start looking for a regression. :)

**Pull request merged**